uaclient/system/cpu_type.py

```diff
@@ -0,0 +1,179 @@
+import abc
+import logging
+import re
+
+from uaclient import exceptions, system, util
+
+LOG = logging.getLogger(util.replace_top_level_logger_name(__name__))
+
+
+class BaseCPUTypeCollector(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def collect(self):
+        pass
+
+    def _get_field_value(self, cpu_info: str, re_pattern):
+        re_match = re.search(
+            re_pattern,
+            cpu_info,
+        )
+
+        if re_match:
+            match_groups = re_match.groupdict()
+            return re.sub(r"\s{2,}", " ", match_groups["name"].strip())
+
+        return None
+
+    def _get_fallback_name(
+        self,
+        cpu_info,
+        fallback_fields,
+        re_pattern,
+    ):
+        fallback_name = ""
+
+        for field in fallback_fields:
+            field_value = (
+                self._get_field_value(
+                    cpu_info=cpu_info,
+                    re_pattern=re_pattern.format(field),
+                )
+                or ""
+            )
+
+            fallback_name += "{}:{}".format(
+                field,
+                field_value,
+            )
+
+            if field != fallback_fields[-1]:
+                fallback_name += "; "
```

Comment:
This is here and then there is at least one hardcoded string that is supposed to fit the same format. Can we have a single implementation of the serialization?
```python
def cpu_type_serialize(d: Dict) -> str:
    # turn the dict into the string
```

and then use that function for all cases where we're returning the serialized string?

Reply:
Just double checking, you want us to first build the dictionary of fallback fields and just use this function to consolidate the dict into a string ?

Reply:
Yes, because fundamentally what we're doing is serializing key-value pairs into a string (though "serialize" is a strong word here because it's not 100% reversible). So we should have a clear single implementation of that serialization that is tested and used anywhere it is needed (including in this function).

Reply:
s/serialize/stringify/